internal/dataplane/translator/translate_tcproute_test.go

```diff
@@ -337,6 +337,157 @@ func TestIngressRulesFromTCPRoutesUsingExpressionRoutes(t *testing.T) {
+							{
+								BackendRefs: []gatewayapi.BackendRef{
+									builder.NewBackendRef("service1").WithPort(80).Build(),
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedKongServices: []kongstate.Service{
+				{
+					Service: kong.Service{
+						Name:     kong.String("tcproute.default.tcprule-1.0"),
+						Protocol: kong.String("tcp"),
+					},
+					Backends: []kongstate.ServiceBackend{
+						{
+							Name:    "service1",
+							PortDef: kongstate.PortDef{Mode: kongstate.PortModeByNumber, Number: int32(80)},
+						},
+					},
+				},
+			},
+			expectedKongRoutes: map[string][]kongstate.Route{
+				"tcproute.default.tcprule-1.0": {
+					{
+						Route: kong.Route{
+							Name:       kong.String("tcproute.default.tcprule-1.0.0"),
+							Expression: kong.String("net.dst.port == 80"),
+							Protocols:  kong.StringSlice("tcp"),
+						},
+						ExpressionRoutes: true,
+					},
+				},
+			},
+		},
+		{
+			name: "single TCPRoute with specific SectionName and Port but they do not match",
+			gateways: []*gatewayapi.Gateway{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "gateway-1",
+						Namespace: "default",
+					},
+					Spec: gatewayapi.GatewaySpec{
+						Listeners: []gatewayapi.Listener{
+							builder.NewListener("tcp80").WithPort(80).TCP().Build(),
+							builder.NewListener("tcp443").WithPort(443).TCP().Build(),
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "gateway-1",
+						Namespace: "test-1",
+					},
+				},
+			},
+			tcpRoutes: []*gatewayapi.TCPRoute{
+				{
+					TypeMeta: tcpRouteTypeMeta,
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "rule-1",
+						Namespace: "default",
+					},
+					Spec: gatewayapi.TCPRouteSpec{
+						CommonRouteSpec: gatewayapi.CommonRouteSpec{
+							ParentRefs: []gatewayapi.ParentReference{
+								{
+									Name:        "gateway-1",
+									SectionName: lo.ToPtr(gatewayapi.SectionName("tcp80")),
+									Port:        lo.ToPtr(gatewayapi.PortNumber(8080)),
+								},
+							},
+						},
+						Rules: []gatewayapi.TCPRouteRule{
+							{
+								BackendRefs: []gatewayapi.BackendRef{
+									builder.NewBackendRef("service1").WithPort(80).Build(),
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedKongServices: []kongstate.Service{
+				{
+					Service: kong.Service{
+						Name:     kong.String("tcproute.default.rule-1.0"),
+						Protocol: kong.String("tcp"),
+					},
+					Backends: []kongstate.ServiceBackend{
+						{
+							Name:    "service1",
+							PortDef: kongstate.PortDef{Mode: kongstate.PortModeByNumber, Number: int32(80)},
+						},
+					},
+				},
+			},
+			expectedKongRoutes: map[string][]kongstate.Route{
+				"tcproute.default.rule-1.0": {
+					{
+						Route: kong.Route{
+							Name:       kong.String("tcproute.default.rule-1.0.0"),
+							Expression: kong.String(""),
+							Protocols:  kong.StringSlice("tcp"),
+						},
+					},
+				},
+			},
+		},
```

Comment:
Can you add a test case with a `TCPRoute` that specifies only the port? the listener should be attached even if no `sectionName` is specified, correct?

And the same goes for the `UDPRoute` tests

Reply:
Yes, correct.

According to [GEP-957 ](https://gateway-api.sigs.k8s.io/geps/gep-957/), there is an example for this scenario.

>The following example shows how an HTTPRoute could be applied to port 8000. In this example, the HTTPRoute will be attached to listeners foo and bar on port 8000 but not listener baz on port 8080.

```yaml
kind: HTTPRoute
metadata:
  name: example
  namespace: example
spec:
  parentRef:
  - name: my-gateway
    port: 8000
  ...
---
kind: Gateway
metadata:
  name: my-gateway
  namespace: example
spec:
  listeners:
  - name: foo
    port: 8000
    protocol: HTTP
    ...
  - name: bar
    port: 8000
    protocol: HTTP
    ...
  - name: baz
    port: 8080
    ...
```